app/controllers/submissions_controller.rb

```diff
@@ -365,7 +369,17 @@ def update_files
                                 ' for assignment ' +
                                 "'#{@assignment.short_identifier}'.")
         end
-
+        #
+        unless students_filename.length < 1
+          if required_assignments != students_filename &&
+             @assignment.is_required == true
+            @file_manager_errors[:size_conflict] =
+            'The list of uploaded files does not match' +
```

Comment:
Please make this an internationalized string (i.e., make a new value in en.yml, fr.yml, and pt.yml). You can use Google translate for the other languages.